concent_api/core/payments/bankster.py

```diff
@@ -474,3 +477,18 @@ def discard_claim(deposit_claim: DepositClaim) -> bool:
         subtask_id=deposit_claim.subtask_id,
     )
     return claim_removed
+
+
+def validate_list_of_transaction_timestamp(
+        list_of_transactions: List[BatchTransferEvent],
+        acceptances: List[SubtaskResultsAccepted],
+) -> None:
+    """ Validate if cut_off_time < T2 + PDT, unless requestor already made some payment. """
+
+    cut_off_time = get_current_utc_timestamp()
+
+    youngest_payments_ts = max(subtask_results_accepted.payment_ts for subtask_results_accepted in acceptances)
+
+    if not any(youngest_payments_ts <= closure_time for closure_time in list_of_transactions):
+        if cut_off_time < youngest_payments_ts + settings.PAYMENT_DUE_TIME:
+            raise BanksterTimestampError
```

Comment:
TODO from https://github.com/golemfactory/concent/pull/1012. Please add logging in this function

Reply:
Done